include/receiver_cadmium.hpp

```diff
@@ -0,0 +1,167 @@
+/**
+ *\brief This is the header file for receiver model.
+ *
+ * Detailed Description : The receiver receives the data and send
+ * back acknowledgment after a fixed time delay. The receiver basically
+ * has two phases, i.e active and passive. Initially the receiver will
+ * be in passive phase and as soon as they receive a packet it changes
+ * to active state and sends the acknowledgment after a time period and
+ * then changes the state to passive.
+ *
+ *
+ */
+
+#ifndef __RECEIVER_HPP__
+#define __RECEIVER_HPP__
+
+
+#include <cadmium/modeling/ports.hpp>
+#include <cadmium/modeling/message_bag.hpp>
+#include <limits>
+#include <math.h> 
+#include <assert.h>
+#include <memory>
+#include <iomanip>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <chrono>
+#include <algorithm>
+#include <limits>
+#include <random>
+
+#include "message.hpp"
+
+using namespace cadmium;
+using namespace std;
+
+/**
+ * This structure contains the input and output messages
+ */
+struct receiver_defs{
+    struct output : public out_port<message_t> {
+    };
+    struct input : public in_port<message_t> {
+    };
+};
+   
+/**
+ *  The class Receiver receives a message and the acknowledgment
+ *  for receiving this message will be passed
+ */
+
+template<typename TIME>
+class Receiver{
+
+	/**< putting definitions in context*/
+    using defs=receiver_defs;
+    public:
+    /**<This constant has the value of time delay between input and output*/
+    TIME   PREPARATION_TIME;
+    /**
+     * Constructor for receiver
+     * It initialize time delay constant with a value
+     */
+    Receiver() noexcept{
```

Comment:
This constructor also initialize ack_number and sending state.
So please add that also in comment.

Reply:
Comment was modified as suggested